e2etests/web/regular_integration_tests/lib/image_loading_main.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  @override
+  MyAppState createState() => MyAppState();
+}
+
+class MyAppState extends State<MyApp> {
+
+  @override
+  void initState() {
+    super.initState();
+    const MethodChannel channel =
+        OptionalMethodChannel('flutter/web_test_e2e', JSONMethodCodec());
+    channel.invokeMethod<void>(
+      'setDevicePixelRatio',
+      '1.5',
+    );
```

Comment:
Part of me says this is working today only because platform messages are synchronous on web. If platform messages ever become async in web, this won't work anymore.

Maybe make `main()` async, and send-and-await this platform message before `runApp()`?

Reply:
+1. Good catch, Mouad. This could cause race conditions down the line. I'm wondering if this falls under the same category as the experiments API, where we use JS-interop instead.

Reply:
Done. Thank you for pointing out reply issue for async.